dataprofiler/profilers/profile_builder.py

```diff
@@ -575,21 +575,54 @@ def _update_profile_from_chunk(df, profile=None, options=None, **kwargs):
         """
         raise NotImplementedError()
 
-    def update_profile(self, data, sample_size, min_true_samples=None):
+    def update_profile(self, data, sample_size=None, min_true_samples=None):
         """
         Update the profile for data provided. User can specify the sample
         size to profile the data with. Additionally, the user can specify the
         minimum number of non-null samples to profile.
 
         :param data: data to be profiled
-        :type data: Union[data_readers.base_data.BaseData, pandas.DataFrame]
+        :type data: Union[data_readers.base_data.BaseData, pandas.DataFrame,
+            pandas.Series]
         :param sample_size: number of samples to profile from the data
         :type sample_size: int
         :param min_true_samples: minimum number of non-null samples to profile
         :type min_true_samples
         :return: None
         """
-        raise NotImplementedError()
+        encoding = None
+        file_type = None
+
+        if isinstance(data, data_readers.base_data.BaseData):
+            encoding = data.file_encoding
+            file_type = data.data_type
+            data = data.data
+        elif isinstance(data, (pd.Series, pd.DataFrame)):
```

Comment:
instead of this, the profilers could have a variable which stipulate what types of frames can be read or a function. this would then allow for these differences.

or the bottom half of the function would be the abstraction and the top would be done by the profilers differently.